src/Functions/reinterpretAs.cpp

```diff
@@ -176,18 +176,23 @@ class FunctionReinterpret : public IFunction
                     size_t size = offsets_from.size();
                     auto & vec_res = col_res->getData();
                     vec_res.resize_fill(size);
+                    size_t copy_size = 0;
```

Comment:
No need to declare `copy_size` and `offset_to` outside the loop, they are both stack variables and assigning something to them isn't performance critical. Feel free to do their definition inside the loop.

Reply:
Updated